ikasaneip/component/validator/src/main/java/org/ikasan/component/validator/xml/XMLValidator.java

```diff
@@ -208,101 +189,88 @@ private String generateErrorMessage(SAXException e, SOURCE source)
         }
     }
 
-    private XMLReader createXMLReader()
-    {
+    private XMLReader createXMLReader() {
         XMLReader reader = null;
-        try
-        {
+        try {
             Class poolClass =
-                    Class.forName("org.apache.xerces.util.XMLGrammarPoolImpl");
+                Class.forName("org.apache.xerces.util.XMLGrammarPoolImpl");
             Object grammarPool = poolClass.newInstance();
             factory.setValidating(true);
             factory.setNamespaceAware(true);
             SAXParser parser = factory.newSAXParser();
             parser.setProperty("http://java.sun.com/xml/jaxp/properties/schemaLanguage",
-                    "http://www.w3.org/2001/XMLSchema");
+                "http://www.w3.org/2001/XMLSchema");
             reader = parser.getXMLReader();
             reader.setErrorHandler(this.errorHandler);
             reader.setProperty(
-                    "http://apache.org/xml/properties/internal/grammar-pool",
-                    grammarPool);
-        }
-        catch (Exception e)
-        {
+                "http://apache.org/xml/properties/internal/grammar-pool",
+                grammarPool);
+            if (configuration.getCatalogUrl() != null) {
+                logger.debug("Setting up Xml Reader with catalog.xml file [{}]", configuration.getCatalogUrl());
+                CatalogResolver cr = CatalogManager.catalogResolver(CatalogFeatures.defaults(), new URI(configuration.getCatalogUrl()));
+                reader.setEntityResolver(cr);
+            }
+        } catch (Exception e) {
             logger.error("Cannot create XMLReader for XSD Validation", e);
             throw new RuntimeException("Cannot create XMLReader for XSD Validation", e);
         }
         return reader;
     }
 
     @Override
-    public String getConfiguredResourceId()
-    {
+    public String getConfiguredResourceId() {
         return configuredResourceId;
     }
 
     @Override
-    public void setConfiguredResourceId(String configuredResourceId)
-    {
+    public void setConfiguredResourceId(String configuredResourceId) {
         this.configuredResourceId = configuredResourceId;
     }
 
     @Override
-    public XMLValidatorConfiguration getConfiguration()
-    {
+    public XMLValidatorConfiguration getConfiguration() {
         return configuration;
     }
 
     @Override
-    public void setConfiguration(XMLValidatorConfiguration configuration)
-    {
+    public void setConfiguration(XMLValidatorConfiguration configuration) {
         this.configuration = configuration;
     }
 
     @Override
-    public void startManagedResource()
-    {
+    public void startManagedResource() {
         this.xmlReaders = new HashMap<Long, XMLReader>();
         // Make sure we can initialise one of these.
-        this.createXMLReader();
+        this.xmlReaders.put(Thread.currentThread().getId(), this.createXMLReader());
```

Comment:
Fixed small buglet here - need to ad the xml reader created to the xmlReaders hashmap